canonical-web-and-design/maas-ui
# feat(ui): filter link modes for the interface type

## Done

- Update the link mode select to limit the options to those valid for the interface type.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- View the network tab for a machine.
- Click add interface.
- Choose a subnet
- Open the IP mode select, you should see 4 options.

## Fixes

Fixes: #2288.
